src/models/CommonModel.ts

```diff
@@ -20,6 +20,102 @@ export class CommonModel extends CommonSchema<CommonModel> {
   extend?: string[];
   originalSchema?: Schema | boolean;
   
+  /**
+   * Retrieves data from originalSchema by given key
+   * 
+   * @param key given key
+   * @returns {any}
+   */
+  getFromSchema<K extends keyof Schema>(key: K) {
+    let schema = this.originalSchema || {};
+    if (typeof schema === 'boolean') schema = {};
+    return schema[`${key}`];
+  }
+
+  /**
+   * Set the types of the model
+   * 
+   * @param types to set the model type to
+   */
+  setType(types : string | string[]) {
+    if (Array.isArray(types)) {
+      if (types.length === 0) {
+        this.type = undefined;
+        return;
+      } else if (types.length === 1) {
+        this.type = types[0];
+        return;
+      }
+    }
+    this.type = types;
+  }
+
+  /**
+   * Adds types to the existing model types.
+   * 
+   * Makes sure to only keep a single type incase of duplicates.
+   * 
+   * @param types which types we should try and add to the existing output
+   */
+  addToTypes(types: string[] | string) {
+    if (Array.isArray(types)) {
+      types.forEach((value) => {
+        this.addToTypes(value);
+      });
+    } else if (this.type === undefined) {
+      this.type = types;
+    } else if (!Array.isArray(this.type) && this.type !== types) {
+      this.type = [this.type, types];
+    } else if (Array.isArray(this.type) && !this.type.includes(types)) {
+      this.type.push(types);
+    }
+  }
+
+  /**
+   * Checks if given property name is required in object
+   * 
+   * @param propertyName given property name
+   * @returns {boolean}
+   */
+  isRequired(propertyName: string): boolean {
+    if (this.required === undefined) {
+      return false;
+    }
+    return this.required.includes(propertyName);
+  }
+
+  /**
+   * This function returns an array of `$id`s from all the CommonModel's it immediate depends on.
+   */
+  getImmediateDependencies(): string[] {
+    const dependsOn = [];
+    if (this.additionalProperties instanceof CommonModel) {
+      const additionalPropertiesRef = (this.additionalProperties as CommonModel).$ref;
+      if (additionalPropertiesRef !== undefined) {
+        dependsOn.push(additionalPropertiesRef);
+      }
+    }
+    if (this.extend !== undefined) {
+      for (const extendedSchema of this.extend) {
+        dependsOn.push(extendedSchema);
+      }
+    }
+    if (this.items instanceof CommonModel) {
```

Comment:
I don't remember, but do we support `items` as array? Should we also iterate on `items` array?

Reply:
Good point, yes! This is actually a bug, gonna create another issue to solve this.

Reply:
https://github.com/asyncapi/generator-model-sdk/issues/186